code/controllers/subsystem/processing/quirks.dm

```diff
@@ -34,7 +33,7 @@ PROCESSING_SUBSYSTEM_DEF(quirks)
 
 /datum/controller/subsystem/processing/quirks/proc/SetupQuirks()
 	// Sort by Positive, Negative, Neutral; and then by name
-	var/list/quirk_list = sortList(subtypesof(/datum/quirk), /proc/cmp_quirk_asc)
+	var/list/quirk_list = sortList(subtypesof(/datum/quirk) - /datum/quirk/item_quirk, /proc/cmp_quirk_asc)
```

Comment:
Could you actually implement an abstract_type variable on /datum/quirk so that you don't need specific code like this?
Then in the loop below, you can do
```dm
if(initial(T.abstract_type) == V)
    continue
```

Then, in your /datum/quirk/item_quirk, simply set abstract_type to its own type.

Reply:
I could, but I don't feel it's necessary.

This is how we effectively handle removing abstract or unwanted types from typesof/subtypes of across the entire codebase. No additional vars required.

Reply:
Side note, the internal implementation of this technique is the same as `typesof` removing the parent type.

![image](https://user-images.githubusercontent.com/24975989/121773519-5ce3a600-cb74-11eb-90a3-f45859b55152.png)

`subtypesof(/datum/foo)` is equivalent to `typesof(/datum/foo) - /datum/foo)` when you get down to the brass tacks.

Reply:
I know but there's a big problem with this implementation where people start doing typesof(/datum/foo) - /datum/foo/type - /datum/foo/other_type - /datum/foo/other_other_type, etc
You make it really hard for people to create abstract types if you don't implement it this way and in the end, it's really just snowflake code.

This is how configs do it and it's very effective.

Reply:
![image](https://user-images.githubusercontent.com/37270891/121775590-991d0380-cb80-11eb-88c7-f7ae450a05f0.png)

Reply:
I tend to agree with Watermelon on this one.
This is not hot code that requires some sort of optimization.
It's much cleaner not to do addition and then subtraction logic.

Reply:
I still disagree that the best solution to this is to add a var that is only useful to quirk type paths that aren't instantiated. It's pointless and unnecessary bloat to add a var that is genuinely only used by datums that aren't instantiated while sitting unused on datums that are.

I feel a more appropriate solution would be to use #defines such as `#define typesof_quirk subtypesof(/datum/quirk) - /datum/quirk/abstract_type_goes_here` which prevent code duplication and allow coders to update these things in only a single place. I would argue that is a more maintainable solution.

However, since more than one maintainer thinks adding additional vars like this is a better solution, I have made this change.